tests/test_generate_data.py

```diff
@@ -3,12 +3,653 @@
+      surface specialized antigen capture cells called [microfold
+      cells](microfold_cell "wikilink") (M cells) that allow for the uptake of
+      antigens produced by pathogens. These M cells then alert the B cells and T
+      cells in the tonsil that a pathogen is present and an immune response is
+      stimulated.[3] B cells are activated and proliferate in areas called
+      germinal centers in the tonsil. These germinal centers are places where B
+      memory cells are created and [secretory antibody (IgA)](Immunoglobulin_A
+      "wikilink") is produced.
+
+    questions_and_answers:
+      - question: |
+          What are the specialized antigen capture cells on the surface of the
+          tonsils called?
+        answer: |
+          The specialized antigen capture cells on the surface of the tonsils
+          are called microfold cells (M cells).
+      - question: What is the role of microfold cells in the tonsils?
+        answer: |
+          Microfold cells (M cells) allow for the uptake of antigens produced by
+          pathogens. They alert the B cells and T cells in the tonsil that a
+          pathogen is present, stimulating an immune response.
+      - question: Where do B cells proliferate in the tonsils?
+        answer: B cells proliferate in areas called germinal centers in the tonsils.
+
+  - context: |
+      A [tonsillolith](tonsillolith "wikilink") (also known as a "tonsil stone")
+      is material that accumulates on the palatine tonsil. This can reach the
+      size of a [peppercorn](peppercorn "wikilink") and is white or cream in
+      color. The main substance is mostly [calcium](calcium "wikilink"), but it
+      has a strong unpleasant odor because of [hydrogen
+      sulfide](hydrogen_sulfide "wikilink") and [methyl
+      mercaptan](methyl_mercaptan "wikilink") and other chemicals.[6]
+
+    questions_and_answers:
+      - question: What is a tonsillolith?
+        answer: |
+          A tonsillolith (tonsil stone) is material that accumulates on the
+          palatine tonsil, reaching the size of a peppercorn and having a white
+          or cream color. It contains calcium and has a strong unpleasant odor
+          due to hydrogen sulfide, methyl mercaptan, and other chemicals.
+      - question: What is the main substance found in a tonsillolith?
+        answer: The main substance found in a tonsillolith is mostly calcium.
+      - question: Why do tonsilloliths have a strong unpleasant odor?
+        answer: |
+          Tonsilloliths have a strong unpleasant odor due to hydrogen sulfide,
+          methyl mercaptan, and other chemicals.
+
+document_outline: |
+  Overview of Human tonsils, describing their types, locations, structure,
+  function, and clinical significance, with a specific focus on their role in
+  the immune system and related health issues.
+
+document:
+  repo: https://github.com/luke-inglis/il-anatomy-knowledge
+  commit: cc7c6ca
+  patterns:
+    - anatomy1.md
+"""
```

Comment:
can I suggest we add these as files in a `mock_data` directory, perhaps?

Reply:
Good point, thanks @khaledsulayman

Reply:
Updated to now load skills from test data